lib/matplotlib/figure.py

```diff
@@ -1113,9 +1113,29 @@ def legend(self, *args, **kwargs):
             # extra_args = extra_args[1:]
             pass
         transform = kwargs.pop('bbox_transform', self.transSubfigure)
+        outside = None
+        if 'loc' in kwargs:
+            # need to protect this in case loc was passed in
+            # positionally...
+            loc = kwargs.pop('loc', None)
+            # handle outside legends:
+            if isinstance(loc, str):
+                if loc.split()[0] == 'outside':
+                    loc = loc.replace('outside ', '')
+                    outside = loc.replace('center ', '')
```

Comment:
Why are we removing "center"?

Reply:
Because we have 'center right' and 'center left' (as opposed to 'left center' and 'right center' which would follow the idiom here), and this allows those to keep working as expected.

Reply:
Added comments